ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/service/AEMDownloadManger.java

```diff
@@ -225,8 +225,17 @@ void extractAemImportsFromManifestsTask() {
                 }
 
                 // add AEM imports extracted from the manifest to the AEM article cache
-                final Manifest manifest = Futures.getUnchecked(mManifestManager.getManifest(translation));
-                repository.addAemImports(translation, manifest.getAemImports());
+                try {
+                    final Manifest manifest = mManifestManager.getManifest(translation).get();
+                    repository.addAemImports(translation, manifest.getAemImports());
+                } catch (final InterruptedException e) {
+                    // set interrupted and return immediately
+                    Thread.currentThread().interrupt();
+                    return;
+                } catch (final ExecutionException e) {
+                    Timber.tag(TAG)
+                            .d(e.getCause(), "Unable to add aem imports from manifest");
+                }
```

Comment:
Just curious why not do a return for the second Catch Statement.  Is there a reason you want this to continue.

Reply:
yes, there are other translations that may still be processed within this method. the `InterruptedException` is special because it means something else is trying to stop the thread executing, and in that scenario you want to exit as quickly as you safely can